optionlab/engine.py

```diff
@@ -35,31 +33,36 @@
 
 
 class StrategyEngine:
-    def __init__(self, inputs: Inputs):
+    def __init__(self, inputs_data: Inputs | dict):
         """
         __init__ -> initializes class variables.
 
         Returns
         -------
         None.
         """
-        self.__s = array([])
-        self.__s_mc: np.ndarray = array(inputs.array_prices or [])
-        self.__strike: list[float] = []
-        self.__premium: list[float] = []
-        self.__n: list[int] = []
-        self.__action: list[Action | Literal["n/a"]] = []
-        self.__type: list[StrategyType] = []
-        self.__expiration: list[dt.date | int] = []
-        self.__prevpos: list[float] = []
-        self.__usebs: list[bool] = []
-        self.__profitranges: list[Range] = []
-        self.__profittargrange: list[Range] = []
-        self.__losslimitranges: list[Range] = []
-        self.__days2maturity: list[int] = []
-        self.__days2target = 30
-        self.__daysinyear = 252 if inputs.discard_nonbusiness_days else 365
-        self.implied_volatility: list[float] = []
+        inputs = (
```

Comment:
That was my previous question. What if the input is neither a Python dictionary nor an Input object, but a JSON string? I know that it is quite simple to transform a JSON string into a Python dictionary using methods of the json module, but I think this is something simple to be already implemented in the lib, such that the user does not need to do that if using optionlab in a pipeline where it exchanges data as JSON strings.

Reply:
See my previous reply. Not a big fan